### ISSUE TITLE
possibly use designmatrix package

### ISSUE THREAD

**Issue by dashaub** [ellisp/forecastxgb-r-package#8]
It looks like you've already setup the coviates to feed into xgboost using lagged values of the timseries--a sensible approach. It could also make sense to included fixed effects for day of week/month/etc.

I started building a package `designmatrix` a few years back to generate `xreg` values to feed into forecasting models and anticipated using it with `forecastHybrid` eventually. It is barely off the ground, but the basic idea is to make it easy to generate covariates for day of week, weekend, month, quarter, etc. Eventually interactions and holidays for these would be nice as well.  If you want to import it, it could serve as a good excuse for me to restart and to finish development. Take a look here: https://github.com/dashaub/designmatrix

**Comment by ellisp** [comment]
Yes I think this would help and I'll import this when it's ready.  I think moving holidays (Easter etc) is a must at some point, although of course people can put them in via xreg.

**Comment by dashaub** [comment]
Great. Easter and major holidays should be easy enough to add, perhaps with an argument `majorHolidays = TRUE`. Any other things you'd like to see?

**Comment by wjhrdy** [comment]
On the same note I have made a package to generate xregs using a list of events and start and end times. 

https://github.com/republicwireless-open/foregen

Considering that most of the effort of forecasting is feature generation if we can speed up that part we could speed up the overall forecasting process by a lot.

**Comment by dashaub** [comment]
And another option, the ["timekit"](https://business-science.github.io/timekit/) package, specifically the `tk_get_timeseries_signature()` function.
